TESTING.md

```diff
@@ -0,0 +1,68 @@
+Below is a draft of the contents, what do y'all think:
+---
+- [General information](#general-information)
+- [Requirements](#requirements)
+- [Running tests](#running-tests)
+  - [Unit tests](#unit-tests)
+  - [Integration tests](#integration-tests)
+  - [Functional tests](#functional-tests)
+- [Continuous Integration](#continuous-integration)
+- [Environment misc](#environment-misc) 
+- [Misc](#misc) 
+
+# General information
+OpenSearch Dashboards uses [Jest](https://jestjs.io/) for unit and integration tests and [Selenium](https://www.selenium.dev/) for functional tests.
+
+In general, we recommend three tiers of tests:
+* Unit tests: Unit tests: small and modular tests that utilize mocks for external dependencies.
+* Integration tests: higher-level tests that verify interactions between systems (eg. HTTP APIs, OpenSearch API calls, calling other plugin). 
+* End-to-end tests (e2e): functional tests that verify behavior in a web browser.
+
+# Requirements
+* Install the latest NodeJS, [NPM](https://www.npmjs.com/get-npm) and [Yarn](https://classic.yarnpkg.com/en/docs/install/#mac-stable)
```

Comment:
Do we actually need to explicitly install npm? I haven't needed to do so.

Reply:
Will merge in now since the page 404s but if we can see if people have issues with this. If so we can update this section.